app/controllers/projects_controller.rb

```diff
@@ -129,6 +129,10 @@ def show_tree_content
 
   def show
     @images = []
+    if @project.blank?
+    redirect_to dashboard_path
+    flash[:alert] = 'The project does not exist.'
+    else
```

Comment:
Small nitpick - fix the indentation of the nested statements, so it's a little more readable.